tola/util.py

```diff
@@ -418,8 +422,9 @@ def ona_parse_type_repeat(data, form_data, parent_name, silo, read):
                     try:
                         entry[field['label']] = entry.pop(parent_name + field['name'])
                     except KeyError as e:
-                        logger.warning(e)
-
+                        warnings.add("Warn: ona_parse_type_repeat for silo %s, %s" % (silo.pk, e))
+    for warn in warnings:
```

Comment:
I don't get the purpose of creating a set to add warnings instead of throwing them at run time.

Reply:
The log was getting flooded with identical warnings, so I thought this was a safe way to eliminate duplicates from the log.  I'd be happy to change it to something else, or revert.  I could just use a boolean to flag when there is a key error and issue a generic warning.  Or add some info to the warning so each one is unique.  It's been a while since I last looked at this so I can't remember what the warning was.